src/lua/api-gateway/redis/redisConnectionProvider.lua

```diff
@@ -0,0 +1,96 @@
+-- Copyright (c) 2017 Adobe Systems Incorporated. All rights reserved.
+--
+--   Permission is hereby granted, free of charge, to any person obtaining a
+--   copy of this software and associated documentation files (the "Software"),
+--   to deal in the Software without restriction, including without limitation
+--   the rights to use, copy, modify, merge, publish, distribute, sublicense,
+--   and/or sell copies of the Software, and to permit persons to whom the
+--   Software is furnished to do so, subject to the following conditions:
+--
+--   The above copyright notice and this permission notice shall be included in
+--   all copies or substantial portions of the Software.
+--
+--   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+--   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+--   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+--   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+--   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+--   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+--   DEALINGS IN THE SOFTWARE.
+
+-- redis endpoints are assumed to be global per GW node and therefore are read here
+
+local restyRedis = require "resty.redis"
+local RedisHealthCheck = require "api-gateway.redis.redisHealthCheck"
+
+local redisHealthCheck = RedisHealthCheck:new({
+    shared_dict = "cachedkeys"
+})
+
+local max_idle_timeout = 30000
+local pool_size = 100
+
+local RedisConnectionProvider = {}
+
+function RedisConnectionProvider:new(o)
+    local o = o or {}
+    self.redis_RO_upstream = self.redis_RO_upstream or "api-gateway-redis-replica"
```

Comment:
this should be back into BasicValidator for that Validators extend BasicValidator and set the properties. This self redis_ro_upstream is never overwritten